numalogic/registry/dynamodb_registry.py

```diff
@@ -0,0 +1,399 @@
+import logging
+import time
+from typing import Any, Optional
+from datetime import datetime, timedelta
+
+import boto3
+from botocore.exceptions import ClientError
+
+from numalogic.registry import ArtifactManager, ArtifactCache, ArtifactData
+from numalogic.tools.exceptions import DynamoDBRegistryError
+from numalogic.tools.types import KEYS, artifact_t, META_VT
+from numalogic.registry._serialize import loads, dumps
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class AWSConnector:
+    """
+    Class to connect to any AWS resource.
+
+    Args:
+        role: AWS role
+        region: AWS region (default: us-west-2)
+    """
+
+    __slots__ = ("role", "region", "session", "sess_start_time")
+
+    __STS_RESET_SECS = 30 * 60
+
+    def __init__(self, role=None, region="us-west-2"):
```

Comment:
```suggestion
    def __init__(self, role: str, region: str = "us-west-2"):
```
I guess role is needed all the time?